src/Core/Query/Helper.php

```diff
@@ -101,7 +101,28 @@ public function escapePhrase(string $input): string
     }
 
     /**
-     * Format a date to the expected formatting used in SOLR.
+     * @deprecated use Helper::convertAndFormatDate
+     *
+     * @codeCoverageIgnore
+     *
+     * @param int|string|\DateTimeInterface $input accepted formats: timestamp, date string, DateTime or
+     *                                             DateTimeImmutable
+     *
+     * @return string|bool false is returned in case of invalid input
+     */
+    public function formatDate($input): string
+    {
+        @trigger_error('The Helper::formatDate function is deprecated in Solarium 6.1 and will be removed in Solarium 6.2. Use Helper::convertAndFormatDate instead', E_USER_DEPRECATED);
+
+        return $this->convertAndFormatDate($input);
+    }
+
+    /**
+     * Convert and format a date to the expected formatting used in SOLR.
+     *
+     * Be aware this method adds a UTC TimeZone to the resulting date which may modify the date time value depending on
```

Comment:
Be aware that this method converts the input into UTC time zone as every dates have to be stored in UTC within Solr. So you might see that the date time "looks" different in the index and is "shifted" by some hours. But this is the correct way to handle different timezones in one index. But it is essential that you provide the correct TZ parameter when you do "date math" in a query. When you get a search result PHP could perfectly convert the date back into the time zone your application requires.
If you want to avoid that conversion you can use Helper::dateToSolrUtcString to format your date without it and just pretend that it is UTC.

Reply:
i'm not sure telling userland how to use dates in solr is helpful?